modules/mgclient.mjs

```diff
@@ -0,0 +1,1799 @@
+     * @return {int}. Memory address of the underlyine object.
+     */
+    transferToWasm() {
+        return this.#cPtr;
+    }
+};
+
+export class MgPoint3D {
+    #cPtr;
+    constructor(cPtr) {
+        this.#cPtr = cPtr;
+    }
+
+    srid() {
+        let wrappedFun = instance.cwrap('mg_point_3d_srid', 'i32', ['i32']);
+        return wrappedFun(this.#cPtr);
+    }
+
+    x() {
+        let wrappedFun = instance.cwrap('mg_point_3d_x', 'i32', ['i32']);
+        return wrappedFun(this.#cPtr);
+    }
+
+    y() {
+        let wrappedFun = instance.cwrap('mg_point_3d_y', 'i32', ['i32']);
+        return wrappedFun(this.#cPtr);
+    }
+
+    z() {
+        let wrappedFun = instance.cwrap('mg_point_3d_z', 'i32', ['i32']);
+        return wrappedFun(this.#cPtr);
+    }
+
+    /**
+     * Returns a copy of the MgPoint3D (allocates memory on the C runtime)
+     * @return {MgPoint3D, null}. null is returned if the allocation fails.
+     */
+    copy(mgPoint3d) {
+        let wrappedFun = instance.cwrap('mg_point_3d_copy', 'i32', ['i32']);
+        let result = new MgPoint3D(wrappedFun(mgPoint2d.transferToWasm()));
+        return errorOrPushToResourceManager(result);
+    }
+
+    /**
+     * Destructs the memory allocated by the C runtime.
+     * This function is used by the ResourceManager to destruct at the end of execution.
+     */
+    destroy() {
+        let wrappedFun = instance.cwrap('mg_point_3d_destroy', 'void', ['i32']);
+        wrappedFun(this.#cPtr);
+    }
+
+    /**
+     * This function shall not be used by this module's consumers.
+     *
+     * Internal function to pass the MgValue to the underlyine wasm mgclient.
+     * @return {int}. Memory address of the underlyine object.
+     */
+    transferToWasm() {
+        return this.#cPtr;
+    }
+};
+
+/**
+ * Helper class to dealloacate resources in one call.
+ */
+class MemoryHandle {
+    #memArray;
+    constructor() {
+        this.#memArray = [];
+    }
+
+    alloc(size) {
+        let ptr = instance._malloc(size);
+        this.#memArray.push(ptr);
+        return ptr;
+    }
+
+    deallocAll() {
+        // deallocate in reverse order
+        this.#memArray.slice().reverse().forEach(element => {
+            instance._free(element);
+        });
+    }
+};
+
+
+/**
+ * Main class used by the module consumers.
+ * This class is the interface between the jsclient and Memgraph database.
+ */
+export class MgClient {
+    #sessionPtr;
+    constructor(sessionPtr) {
+        this.#sessionPtr = sessionPtr;
+    }
+
+    /**
+     * This function connects the client to the memgraph instance. On success returns an object of this class.
+     * On failure it returns null.
+     * @param {string} host. The host address of the memgraph instance.
+     * @param {port} port. The port address of the memgraph instance.
+     * @return {MgClient, null}.
+     */
+    static async connect(host, port) {
+        // todo expose ssl
```

Comment:
Definitely expose SSL here 😄

Reply:
This shall be on a patch release. I haven't even enabled it on `mgclient` :)